Bottom: 288c9cf209f61ed951b03bd51dbf911e9ad9044a
Top:    ef5a60e573027f22c928b8eb4e5e46cb5154b3ab
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-10 20:31:45 +0000

Try not using no-holes dicer for drawing polygons











---

diff --git a/src/draw.c b/src/draw.c
index f282b44..0384f62 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -2169,6 +2169,7 @@ DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon)
     }
   else if (Polygon->Clipped)
     {
+#if 0
       if (!Polygon->NoHolesValid)
         {
           ComputeNoHoles (Polygon);
@@ -2182,6 +2183,15 @@ DrawPlainPolygon (LayerTypePtr Layer, PolygonTypePtr Polygon)
             poly.Clipped = poly.Clipped->f;
           } while (poly.Clipped != Polygon->NoHoles);
         }
+#endif
+        {
+          PolygonType poly = *Polygon;
+          poly.Clipped = Polygon->Clipped;
+          do {
+            DrawPolygonLowLevel (&poly, NULL);
+            poly.Clipped = poly.Clipped->f;
+          } while (poly.Clipped != Polygon->Clipped);
+        }
       /* draw other parts of the polygon if fullpoly flag is set */
       /* NB: No "NoHoles" cache for these */
       if (TEST_FLAG (FULLPOLYFLAG, Polygon))
